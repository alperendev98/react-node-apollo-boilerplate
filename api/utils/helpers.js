function getValidationRules(allowGraphqlIntrospection, graphqlDisableIntrospection) {
	if (!allowGraphqlIntrospection && !graphqlDisableIntrospection) return [];

	return allowGraphqlIntrospection ? [] : [graphqlDisableIntrospection];
}

module.exports = {
	getValidationRules
};
