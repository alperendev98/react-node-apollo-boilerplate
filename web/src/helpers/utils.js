import pickBy from 'lodash/pickBy';
import isObject from 'lodash/isObject';
import isArray from 'lodash/isArray';

export function processModifiers(modifiers) {
	return isObject(modifiers) && !isArray(modifiers) ? pickBy(modifiers) : modifiers;
}
