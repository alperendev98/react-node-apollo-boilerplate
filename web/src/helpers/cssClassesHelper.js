import BEMHelper from 'react-bem-helper';
import { processModifiers } from './utils';

export const getBEMClasses = (...classes) => {
	classes = Array.prototype.concat(...classes).filter(item => item);

	const BEMClasses = classes.map(
		className =>
			new BEMHelper({
				name: className,
				outputIsString: true
			})
	);
	return (element, modifier) =>
		BEMClasses.map(BEMClass => BEMClass(element, processModifiers(modifier))).join(' ');
};

export const addModifier = (modifiers, newModifier) => {
	let result;

	if (Array.isArray(modifiers)) {
		result = [newModifier, ...modifiers];
	} else if (typeof modifiers === 'string') {
		result = [newModifier, modifiers];
	} else if (typeof modifiers === 'object') {
		result = modifiers;
		result[newModifier] = true;
	}

	return result;
};
