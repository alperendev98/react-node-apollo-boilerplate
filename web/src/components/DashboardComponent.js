import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import { getBEMClasses } from '../helpers/cssClassesHelper';
import UtilityContainer from '../containers/UtilityContainer';
import EquipmentContainer from '../containers/EquipmentContainer';

import '../styles/components/dashboardComponent.css';

class DashboardComponent extends React.Component {
	render() {
		return (
			<React.Fragment>
				<Switch>
					<Route path="/admin/utilities" component={UtilityContainer} />
					<Route path="/admin/equipment" component={EquipmentContainer} />
					<Redirect to="/admin/utilities" />
				</Switch>
			</React.Fragment>
		);
	}
}

export default DashboardComponent;
