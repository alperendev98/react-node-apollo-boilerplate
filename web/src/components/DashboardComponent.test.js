import React from 'react';
import { shallow } from 'enzyme';
import { getBEMClasses } from '../helpers/cssClassesHelper';
import DashboardComponent from './DashboardComponent';

describe('DashboardComponent.js TESTS', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<DashboardComponent />);
	});

	it('renders DashboardComponent without crashing', () => {
		expect(wrapper.length).toEqual(1);
	});

	it('should match previos snapshot for DashboardComponent', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
