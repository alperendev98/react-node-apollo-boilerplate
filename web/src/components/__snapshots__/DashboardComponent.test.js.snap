// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardComponent.js TESTS should match previos snapshot for DashboardComponent 1`] = `
<Fragment>
  <Switch>
    <Route
      component={[Function]}
      path="/admin/utilities"
    />
    <Route
      component={[Function]}
      path="/admin/equipment"
    />
    <Redirect
      push={false}
      to="/admin/utilities"
    />
  </Switch>
</Fragment>
`;
